samples/panther/ap/src/ipm.c

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (c) 2017 IpTronix S.r.l.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <stdio.h>
+#include <ipm.h>
+#include <ipm/ipm_quark_se.h>
+#include <sensor.h>
+#include <zephyr.h>
+#include <misc/printk.h>
+
+#include <mqtt.h>
+
+#include "mqtt.h"
+
+#define STACKSIZE               2000
+
+#define MAIN_FIBER_PRI          2
+
+struct shield_sens {
+	struct sensor_value BME280[3];
+	struct sensor_value ADXL362[4];
+	struct sensor_value ADXRS290[3];
+	struct sensor_value SI1153[5];
+};
+
+QUARK_SE_IPM_DEFINE(ess_ipm, 0, QUARK_SE_IPM_INBOUND);
+
+static char ipm_stack[STACKSIZE];
+static char json[2048];
+static struct shield_sens sensors;
+static struct k_sem	sync_sem;
+static char *mqtt_device_hid;
+static char *mqtt_topic;
+
+static void sensor_ipm_callback(void *context, uint32_t id, volatile void *data);
+static void ipm_thread(void *arg1, void *arg2, void *arg3);
+static int json_get(void);
+
+int ipm_init(char *device_hid, char *topic)
+{
+	struct device *ipm;
+	int ret;
+
+	mqtt_device_hid = device_hid;
+	mqtt_topic = topic;
+
+	ipm = device_get_binding("ess_ipm");
+	if (!ipm) {
+		return -1;
+	}
+
+	ipm_register_callback(ipm, sensor_ipm_callback, NULL);
+
+	ret = ipm_set_enabled(ipm, 1);
+	if (ret) {
+		return -1;
+	}
+
+	k_sem_init(&sync_sem, 0, 1);
+
+	k_thread_spawn(ipm_stack, STACKSIZE, ipm_thread,
```

Comment:
The k_thread_spawn() is deprecated, you should use k_thread_create() instead.

Note also that PR #392 (not yet merged) will bring some changes how the stack is defined.